### PR TITLE
Fix P2P SDP negotiation with non-human-role clients

### LINKED ISSUE #359: Automatic cloud recording not capturing participants who join after recorder
Reported from customer: 

> We’re using @whereby.com/browser-sdk and want to auto start cloud recording. From our observation, the participants joining the room after a certain time do not get their video recorded and their videos appears as spinners in the recording. The steps are as follows:
> 
> 1) 1st participant joins the room <— video recorded
> — 2 seconds elapsed and get a new_client msg of role recorder from wss://signal.appearin.net/protocol/socket.io/v4 —
> 2) 2nd participant joins the room <— video not recorded
> 3) 3rd participant joins the room <— video not recorded

> We’re using Whereby’s React SDK to implement our custom video conference solution
> - the meetingUrl is passed to the useRoomConnection hook and I observed the said issue on every meetingUrl.
> - The issue did not happen if the participants directly join via the meetingUrl in the browser.
> - In every test where some guests were not recorded, all guests were able t see each other on the live call (implemented by VideoView  in the React SDK), but guests who were not recorded appears as spinners in the recording file.

### PR DESCRIPTION
**Pull request by pnts-se-whereby**
### Description

**Summary:**

We need to keep remote clients with non-human-roles in our remote state. For instance, we need to ask them for an SDP offer in P2P or a PeerConnection wont be established.

closes #359 

**Related Issue:**

### Checklist

-   [x] My code follows the project's coding standards.
-   [x] Prefixed the PR title and commit messages with the service or package name
-   [ ] I have written unit tests (if applicable).
-   [ ] I have updated the documentation (if applicable).
-   [x] By submitting this pull request, I confirm that my contribution is made
        under the terms of the MIT license.

### CODE REVIEW

File: packages/core/src/redux/slices/remoteParticipants.ts [hunk 1]
```diff
@@ -211,10 +211,7 @@ export const remoteParticipantsSlice = createSlice({
 
             return {
                 ...state,
-                remoteParticipants: clients
-                    .filter((c) => c.id !== selfId)
-                    .filter((c) => !NON_PERSON_ROLES.includes(c.role.roleName))
-                    .map((c) => createRemoteParticipant(c)),
+                remoteParticipants: clients.filter((c) => c.id !== selfId).map((c) => createRemoteParticipant(c)),
```
Comment (nandito) [inline packages/core/src/redux/slices/remoteParticipants.ts:214]: Isn't this a breaking change? Some clients probably need to tweak their app to exclude non-person clients from their video grid, if this gets merged. 🤔

**Review by havardholvik — COMMENTED** [review]
Are we sure this wont have any unintended side-effects? I know we discussed this @thyal at one point; clearly separating **clients** from **participants**. 

EDIT. I see you already did this on the selector level Thomas. Should we then rename the internal state to `remoteClients` to avoid confusion?

**Review by havardholvik — APPROVED** [review]

File: packages/core/src/redux/slices/remoteParticipants.ts (hunk 1 above)
Comment (nandito) [reply packages/core/src/redux/slices/remoteParticipants.ts:214]: Ah nevermind, as I see we still have the filter in the `selectRemoteParticipants = createSelector(selectRemoteClients, (clients) => {}` fn on line 307 👍

**Review by thyal — COMMENTED** [review]
I think we need to do some additional changes - This will also show non-human participants in the grid f.ex. I'll do some testing now to make sure.

**Review by nandito — APPROVED** [review]
LGTM 👏

File: packages/core/src/redux/slices/remoteParticipants.ts (hunk 1 above)
Comment (havardholvik) [reply packages/core/src/redux/slices/remoteParticipants.ts:214]: Yea. That was the one I noticed as well. So I _think_ it wont break anything.

**Comment by havardholvik** [comment]
> I think we need to some additional changes - This will also show non-human participants in the grid f.ex. I'll do some testing now to make sure.

Are you sure. @thyal ? There is already a selector filtering out non-humans

**Comment by thyal** [comment]
> Are you sure. @thyal ? There is already a selector filtering out non-humans

Ah, yeah I see it now. Then it should be good.

**Comment by nandito** [comment]
It's nice to see that Håvard and Thomas had the same panic as I had 😅 

I tested it with the sample app and the recorder client did not appear as a new grid + it's clear in the code that the `selectRemoteParticipants` excludes non-person roles. The `remoteClients` vs `remoteParticipants` array names are a bit confusing though.

**Comment by thyal** [comment]
> It's nice to see that Håvard and Thomas had the same panic as I had 😅
> 
> I tested it with the sample app and the recorder client did not appear as a new grid + it's clear in the code that the `selectRemoteParticipants` excludes non-person roles. The `remoteClients` vs `remoteParticipants` array names are a bit confusing though.

Yes, it's a bit confusing, but also it makes sense - clients is every client (including non-person), and participants is only human. We can rename the internal state to `remoteClients` as @havardholvik suggested though, that might make it a bit clearer

**Pull request merged**